core/chains/evm/client/errors.go

```diff
@@ -268,6 +268,10 @@ var mantle = ClientErrors{
 	NonceTooLow:     regexp.MustCompile(`(: |^)'*nonce too low`),
 }
 
+var berachain = ClientErrors{	
+	Fatal:           regexp.MustCompile(`(: |^)'*invalid chain ID`),	
```

Comment:
it is indeed coming from the compat tests, particularly the `invalid_sender.go` which is expecting the exact literal "invalid sender" from the error and Bera returns "invalid chain ID".

Will remove it.

Reply:
@simsonraj Can you please clarify why the compat test results is not relevant here?
The same test (approach for reproducing the invalid signer) gives a correct result (invalid signer error) on the other chains, only Berachain fails it.

Reply:
The tests mimic the invalid signature by modifying the chainID, which works for about 50% of the chains. Many other chains respond with "Invalid chain ID" which means they also have an additional validation for this. 

My question was going to be If all the other chains are also OP stack how come only Berachain fails, I just noticed Berachain is not OPstack so it makes sense that they have an additional validation.

Reply:
perfect, so for berachain is good to have this error being mapped.
Closing this one for the time being and if it needs more clarification we can see it later. 
Thanks Simson and Mo.

Reply:
SOrry Im saying opposite, we dont need this error classification, because our core node is never going to go into this scenario to reproduce this error

Reply:
removed this error mapping. closing this one as resolved.